### PR TITLE
add afbc for exposing a "colorize" api to bbs; addresses #8588 

### PR DESCRIPTION
**Pull request by lowlyw**

### CODE REVIEW

File: libr/anal/anal.c [hunk 1]
```diff
@@ -346,6 +346,21 @@ R_API void r_anal_trace_bb(RAnal *anal, ut64 addr) {
 	}
 }
 
+R_API void r_anal_colorize_bb(RAnal *anal, ut64 addr) {
+	RAnalBlock *bbi;
+	RAnalFunction *fcni;
+	RListIter *iter2;
+	fcni = r_anal_get_fcn_in (anal, addr, 0);
+	if (fcni) {
+		r_list_foreach (fcni->bbs, iter2, bbi) {
```
Comment (radare) [inline libr/anal/anal.c]: use `R_API RAnalBlock *r_anal_bb_from_offset(RAnal *anal, ut64 off) {`

File: libr/core/cmd_anal.c [hunk 2]
```diff
@@ -2625,6 +2625,21 @@ static int cmd_anal_fcn(RCore *core, const char *input) {
...
+					RAnalOp *op = r_core_op_anal (core, addr);
+					if (op) {
+						r_anal_colorize_bb (core->anal, addr);
+						r_anal_op_free (op);
+					} else {
+						eprintf ("Cannot analyze opcode at 0x%08" PFMT64x "\n", addr);
+					}
+				}
+			}
+            break;
```
Comment (radare) [inline libr/core/cmd_anal.c]: bad indent

File: libr/include/r_anal.h [hunk 3]
```diff
@@ -1702,6 +1703,8 @@ R_API char *r_anal_rtti_demangle_class_name(RAnal *anal, const char *name);
 R_API void r_anal_rtti_print_at_vtable(RAnal *anal, ut64 addr, int mode);
 R_API void r_anal_rtti_print_all(RAnal *anal, int mode);
 
+R_API void r_anal_colorize_bb(RAnal *anal, ut64 addr);
```
Comment (radare) [inline libr/include/r_anal.h]: add another argument to specify the color, or at least a number. like its done with the traces. See the dt command

**Comment by XVilka** [comment]
Please mention https://github.com/radare/radare2/issues/8588 in the commit message

**Comment by XVilka** [comment]
@lowlyw but why capitalized "H"? Why not just "h"?

**Comment by Maijin** [comment]
`afbH` was making sense I guess because of `ecH` is for `highlight word or instruction`

File: libr/core/cmd_anal.c [hunk 4]
```diff
@@ -2625,6 +2626,25 @@ static int cmd_anal_fcn(RCore *core, const char *input) {
...
 			anal_fcn_add_bb (core, input + 3);
 			break;
+		case 'c': // "afbc"
+			{
+			const char *ptr = input + 3;
+			ut64 addr = r_num_math (core->num, ptr);
+			ut32 color;
+			ptr = strchr (ptr, ' ');
+			if (ptr) {
+				ptr = strchr (ptr + 1, ' ');
```
Comment (MaskRay) [inline libr/core/cmd_anal.c]: bug here `ptr = strchr (ptr + 1, ' ');`

File: libr/core/cmd_anal.c [hunk 5]
```diff
@@ -2625,6 +2626,25 @@ static int cmd_anal_fcn(RCore *core, const char *input) {
 		case '+': // "afb+"
 			anal_fcn_add_bb (core, input + 3);
 			break;
+		case 'c': // "afbc"
+			{
```
Comment (MaskRay) [inline libr/core/cmd_anal.c]: `case 'c': { // "afbc"`

File: libr/core/cmd_anal.c (hunk 4 above)
Comment (radare) [reply libr/core/cmd_anal.c]: cc @lowlyw

File: libr/core/cmd_anal.c (hunk 5 above)
Comment (radare) [reply libr/core/cmd_anal.c]: cc @lowlyw

**Pull request merged**